### PR TITLE
[Filebeat] Rename awscloudwatch input to aws-cloudwatch input

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to rename `awscloudwatch` input to `aws-cloudwatch` to follow the naming convention for Filebeat inputs.
After this PR get cherry-picked into 7.x, we can remove support of `awscloudwatch`. Or create a separate issue to remove old names in 8.0.

Deprecation warning log message is also added to this PR:
```
2020-10-28T12:41:21.556-0600    WARN    [aws-cloudwatch]        awscloudwatch/input.go:93       awscloudwatch input name is deprecated, please use aws-cloudwatch instead
```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Make sure both `awscloudwatch` and `aws-cloudwatch` input work:
```
filebeat.inputs:
  - type: awscloudwatch
    credential_profile_name: elastic-beats
    log_group_arn: arn:aws:logs:us-east-1:428152502467:log-group:test:*
    region: us-east-1
    scan_frequency: 30s
    start_position: beginning
    api_timeout: 5m
```
and 
```
filebeat.inputs:
  - type: aws-cloudwatch
    credential_profile_name: elastic-beats
    log_group_arn: arn:aws:logs:us-east-1:428152502467:log-group:test:*
    region: us-east-1
    scan_frequency: 30s
    start_position: beginning
    api_timeout: 5m
```

## Related issues

- Relates https://github.com/elastic/beats/pull/22214

### CODE REVIEW

File: x-pack/filebeat/input/awscloudwatch/input.go [hunk 1]
```diff
@@ -26,13 +26,21 @@ import (
...
+	oldInputName = "awscloudwatch"
+)
 
 func init() {
 	err := input.Register(inputName, NewInput)
 	if err != nil {
 		panic(errors.Wrapf(err, "failed to register %v input", inputName))
 	}
+
+	err = input.Register(oldInputName, NewInput)
```
Comment (ChrsMark) [inline x-pack/filebeat/input/awscloudwatch/input.go:40]: Maybe to also log that this name will be deprecated?

Comment (ChrsMark) [reply x-pack/filebeat/input/awscloudwatch/input.go:40]: 👍🏼

**Review by ChrsMark — APPROVED** [review]
lgtm!

File: x-pack/filebeat/input/awscloudwatch/input.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/filebeat/input/awscloudwatch/input.go:40]: Yep it's in https://github.com/elastic/beats/pull/22228/files#diff-caf6618b123c7fe852f46f717af459389cfc03b351ca89d8263e6500574993e3R93 :) The output log looks like this: 
```
2020-10-28T12:41:21.556-0600    WARN    [aws-cloudwatch]        awscloudwatch/input.go:93       awscloudwatch input name is deprecated, please use aws-cloudwatch instead
```

**Pull request merged**